engine/client/buildkit.go

```diff
@@ -52,51 +45,11 @@ func newBuildkitClient(ctx context.Context, remote *url.URL, userAgent string) (
 		return nil, fmt.Errorf("buildkit client: %w", err)
 	}
 
-	return c, nil
-}
-
-// waitBuildkit waits for the buildkit daemon to be responsive.
-// TODO: there's a builtin method for this now
-func waitBuildkit(ctx context.Context, host string) ([]*bkclient.WorkerInfo, error) {
-	// Try to connect every 100ms up to 1800 times (3 minutes total)
-	// NOTE: the long timeout accounts for startup time of the engine when
-	// it needs to synchronize cache state.
-	const (
-		retryPeriod   = 100 * time.Millisecond
-		retryAttempts = 6000
-	)
-
-	var c *bkclient.Client
-	var err error
-
-	for retry := 0; retry < retryAttempts; retry++ {
-		c, err = bkclient.New(ctx, host, bkclient.WithFailFast())
-		if err == nil {
-			break
-		}
-		time.Sleep(retryPeriod)
-	}
-
-	if err != nil {
-		return nil, fmt.Errorf("buildkit failed to respond: %w", err)
-	}
-
-	if c == nil {
-		return nil, fmt.Errorf("buildkit failed to respond")
-	}
-
-	// FIXME Does output "failed to wait: signal: broken pipe"
-	defer c.Close()
-
-	var workerInfo []*bkclient.WorkerInfo
-	for retry := 0; retry < retryAttempts; retry++ {
-		workerInfo, err = c.ListWorkers(ctx)
-		if err == nil {
-			return workerInfo, nil
-		}
-		time.Sleep(retryPeriod)
+	ctx, cancel := context.WithTimeout(ctx, 10*time.Minute)
```

Comment:
I let @vito or @sipsma double check but according to the CI, it seems like it perfectly work!